### PR TITLE
Fix node label customisation

### LINKED ISSUE #674: Provision fails when openshift_cluster_node_labels does not define at least one label for each group
## Bug Report

When getting values of `openshift_cluster_node_labels` in [heat_stack.yaml.j2](https://github.com/openshift/openshift-ansible-contrib/blob/master/roles/openstack-stack/templates/heat_stack.yaml.j2#L688) during provisioning, if `openshift_cluster_node_labels.infra` (or `.app`) is not defined, provisioning ends with an error (that `iteritems()` cannot be called for an undefined object).

This happens when someone overrides the  default `openshift_cluster_node_labels` value defined [here](https://github.com/openshift/openshift-ansible-contrib/blob/master/roles/common/defaults/main.yml) with their own values but does not define at least one label for both `app` and `infra` group.

### PR DESCRIPTION
**Pull request by Tlacenka**
#### What does this PR do?
This PR fixes problems with label customisation.

- In order for the heat templates to work, it adds an empty dictionary to both infra and app groups if there isn't any label defined for them.
- The `openshift_cluster_node_labels` definition is moved to `all.yml` due to its usage in the `openstack-stack` role during provisioning.
- Documentation for customising node labels is expanded.

#### How should this be manually tested?
1) In `all.yml`, add custom definition to `openshift_cluster_node_labels` which contains only one of the available groups (app or infra).
2) Run provisioning with `-vvv` option and take a look at the output of the tasks right before the `openstack-stack` role (in particular, `Set openshift_cluster_node_labels for the app group if not defined`). It should show the content of `openshift_cluster_node_labels` variable that has both `app` and `infra` keys in it (along with your custom label).
3) Check that the end to end deployment is successful.
4) Connect to the OpenShift master node and display all nodes, including their labels with `oc get nodes --show-labels`. Your custom labels should be printed out along with the predefined labels.

#### Is there a relevant Issue open for this?
Closes #674 

#### Who would you like to review this?
cc: @tomassedovic @bogdando PTAL

### CODE REVIEW

**Comment by Tlacenka** [comment]
yey, it passed, and so did the build @bogdando started ( https://travis-ci.org/openshift/openshift-ansible-contrib/builds/267579894 )

File: playbooks/provisioning/openstack/pre_tasks.yml [hunk 1]
```diff
@@ -31,3 +31,19 @@
...
   when:
   - openshift_master_default_subdomain is undefined
+
+# Check that openshift_cluster_node_labels has regions defined for all groups
+# NOTE(kpilatov): if node labels are to be enabled for more groups,
+#                 this check needs to be modified as well
+- name: Set openshift_cluster_node_labels if undefined (should not happen)
+  set_fact:
+    openshift_cluster_node_labels: {'app': {'region': 'primary'}, 'infra': {'region': 'infra'}}
+  when: openshift_cluster_node_labels is not defined
```
Comment (bogdando) [inline playbooks/provisioning/openstack/pre_tasks.yml]: An elegant solution to catch up changes for the all.yml for folks who're used to not update it. I like it!

**Comment by tomassedovic** [comment]
@bogdando did you test the features here? I was under the impression that @Tlacenka wanted to finish her local tests before merging this.

**Comment by tomassedovic** [comment]
@bogdando oh my bad, I think that was the scaling up PR. This is fine.

**Comment by bogdando** [comment]
yeah, that one passed CI

**Pull request merged**